### PR TITLE
Update HttpTransportBindingElement with Proxy property to enable client to specify IWebProxy

### LINKED ISSUE #4562: System.InvalidOperationException: The required UserNameSecurityToken was not provided
Configuring proxy with AuthenticationSchemes.Basic (see code) results in error in title: 
Related issue #4557
```
                var customBinding = new CustomBinding(result);

                var htbe = customBinding.Elements.Find<HttpTransportBindingElement>();
                htbe.ProxyAuthenticationScheme = AuthenticationSchemes.Basic; // Or whatever authentication mechanism your proxy server uses
                htbe.BypassProxyOnLocal = true;
                htbe.UseDefaultWebProxy = false;

                htbe.ProxyAddress = new Uri("http://user:pass@someproxy");
```

### PR DESCRIPTION
**Pull request by dimalyshev**
Fix #4562
- update HttpTransportBindingElement with Proxy property 
- update HttpChannelFactory to enable _proxy configuring with bindingElement.Proxy 

this commit enables clients to configure binding with fully specified IWebProxy object:
- address , 
- credentials 
- etc.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/wcf?pullRequest=4563) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/wcf?pullRequest=4563) before we can accept your contribution.<br/><br/>:x: dimalyshev [sign now](https://cla.dotnetfoundation.org/dotnet/wcf?pullRequest=4563)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/wcf?pullRequest=4563) it.</sub>

**Comment by dimalyshev** [comment]
@mconnew  hi again! Sorry for annoying mentioning and pushing,
but have you managed to look it through ?

Thanx in advance for response!

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs [hunk 1]
```diff
@@ -100,7 +100,9 @@ internal HttpChannelFactory(HttpTransportBindingElement bindingElement, BindingC
             TransferMode = bindingElement.TransferMode;
             _keepAliveEnabled = bindingElement.KeepAliveEnabled;
 
-            if (bindingElement.ProxyAddress != null)
+            if (bindingElement.Proxy != null)
+                _proxy = bindingElement.Proxy;
```
Comment (mconnew) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs:104]: Coding style requires if statements to be in a code block
```suggestion
            {
                _proxy = bindingElement.Proxy;
            }
```

**Comment by mconnew** [comment]
This won't make the Proxy property available to you as it's not added to the contract (in src\System.ServiceModel.Http\ref\System.ServiceModel.Http.cs). But we have a general policy of not exposing new api in patch releases so I don't know how to easily solve this for you. Would accessing that property using reflection with the knowledge that it will be added to the public contract in the next major release work for you?

**Comment by dimalyshev** [comment]
Thank you for decision and proposal:
well, I'm not surely undestand using of Ref libraries built - as far as I understand the only aim to satisfy dependensies for other Ref libraries.
Anyway if produced libraries allow to set Proxy property - it is ok!
Let's build nuget? How long will it takes?

**Comment by mconnew** [comment]
We have an implementation assembly/package (System.Private.ServiceModel) and contract packages (eg System.ServiceModel.Http). The contract package contains the reference assembly, this is the assembly you compile against. If the api isn't in the reference assembly, you can't use it in your code even if it's in the implementation assembly (System.Private.ServiceModel). We don't add new api's to contracts in patch releases which means you won't be able to compile against the api even if we ship an updated implementation assembly with it there. Which is why I was asking if you would be okay with using the api via reflection. Otherwise I'll need to look into seeing if an exception is possible.

**Comment by dimalyshev** [comment]
I appreciate your explanation.
I have agreed recently to get nuget with this patch to be able to set Proxy property via reflection
until you'll be  ship updated reference contract.
That's why I've asked - how long does it take to build nuget with patch and publish it to nuget.org?

**Comment by dimalyshev** [comment]
Hello, @mconnew , what is news?
When nuget build shipping is expected

Thanx in advance...

**Comment by mconnew** [comment]
@dimalyshev, this change will be released in a preview release as it's new API. We're still working out when that will be, but it won't be any later than the Build conference that's happening next month.

**Pull request merged**